pkg/query/querier.go

```diff
@@ -365,6 +376,52 @@ func (q *querier) Select(ctx context.Context, _ bool, hints *storage.SelectHints
 }
 
 func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms ...*labels.Matcher) (storage.SeriesSet, storepb.SeriesStatsCounter, error) {
+	if q.enablePreAggregationLabelRewrite {
```

Comment:
please consider move them to a separate function or file so it is easier to merge upstream code and unit test

Reply:
please modualrize the code so it is not a giant piece and please unit test it, this is critical functionality to ensure no regression